File: versioned_docs/version-1.10.x/examples/js/src/data_replication_browse.mjs
import {Client, QuotaType, ReplicationSettings} from "reduct-js";
import assert from "node:assert";

// Create a client instance
const client = new Client("http://127.0.0.1:8383", {apiToken: "my-token"});

// List all replications
for (const replication of await client.getReplicationList()) {
    console.log("Replication: ", replication.name);
    console.log("Active: ", replication.isActive);
    console.log("Pending records: ", replication.pendingRecords);
    console.log("Provisioned: ", replication.isActive);
}

// Get all details of a replication
const replication = await client.getReplication("example-replication");
console.log("Replication: ", replication.info.name);
console.log("Settings: ", replication.settings);
console.log("Failed records (last hour): ", replication.diagnostics.hourly.errored)
console.log("Successful records (last hour): ", replication.diagnostics.hourly.ok)
console.log("Errors (last hour): ", replication.diagnostics.hourly.errors)

File: versioned_docs/version-1.12.x/examples/py/src/data_replication_browse.py
import asyncio

from reduct import Client, ReplicationSettings


async def main():
    # Create a client instance
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
        # List all replications
        for replication in await client.get_replications():
            print("Replication: ", replication.name)
            print("Active: ", replication.is_active)
            print("Pending records: ", replication.pending_records)
            print("Provisioned: ", replication.is_provisioned)

        # Get all details of a replication
        replication = await client.get_replication_detail("example-replication")
        print("Replication: ", replication.info.name)
        print("Settings: ", replication.settings)
        print("Failed records (last hour): ", replication.diagnostics.hourly.errored)
        print("Successful records (last hour): ", replication.diagnostics.hourly.ok)
        print("Errors (last hour): ", replication.diagnostics.hourly.errors)


loop = asyncio.get_event_loop()
loop.run_until_complete(main())
